File: Interpreter/AsinhFun.java
package calculator.xwg;

import java.util.LinkedList;

public class AsinhFun extends CalculateFunction {
	String getName(){
		return "asinh";
	}
	
	boolean execute(LinkedList<Complex> paraList, EvaluateContext context){
		if(paraList.size() != 1)
	    {
			context.setErrorMessage(getName(), R.string.error_invalid_parameter_count);
	        return false;
	    }
	    Complex para = paraList.getFirst();
	    if(para.i != 0)
	    {
	    	context.setErrorMessage(getName(), R.string.error_invalid_date_type);
	        return false;
	    }
	    context.setCurrentResult(new Complex(Math.log(para.r + Math.sqrt(1 + para.r * para.r))));
	    return checkResult(context);
	}
}

File: Interpreter/AsinhFun.py
import math
from Interpreter.CalculateFunction import *
from Interpreter.Complex import *


class AsinhFun(CalculateFunction):
    def getName(self):
        return 'asinh'

    def execute(self, paraList, context):
        if (len(paraList) != 1):
            context.setErrorMessage(self.getName(), R_string.error_invalid_parameter_count)
            return False

        para = paraList[0]
        if (para.i != 0):
            context.setErrorMessage(self.getName(), R_string.error_invalid_date_type)
            return False

        context.setCurrentResult(Complex(math.log(para.r + math.sqrt(1 + para.r * para.r))))
        return self.checkResult(context)
